cppapi/client/devapi_base.cpp

```diff
@@ -2688,6 +2669,37 @@ DeviceProxy::~DeviceProxy()
 #endif
 }
 
+void DeviceProxy::unsubscribe_all_events()
+{
+    if (ApiUtil* api = ApiUtil::instance())
+    {
+        if (ZmqEventConsumer* zmq_event_consumer = api->get_zmq_event_consumer())
+        {
+            vector<int> event_ids;
+            zmq_event_consumer->get_subscribed_event_ids(this, event_ids);
+
+            for (vector<int>::iterator event_id = event_ids.begin();
+                event_id != event_ids.end();
+                ++event_id)
+            {
+                try
+                {
+                    unsubscribe_event(*event_id);
+                }
+                catch (CORBA::Exception& e)
+                {
+                    Tango::Except::print_exception(e);
+                }
+                catch (...)
+                {
+                    std::cerr
+                        << "Unknown exception thrown from unsubscribe_event() event_id="
+                        << *event_id << std::endl;
```

Comment:
I think we should also add the name of the device in the error message as well as the current method name.
So I would suggest to change the error message with something like:
```
std::cerr << " DeviceProxy::unsubscribe_all_events(): Unknown exception thrown from unsubscribe_event() for device \""
             << name() << "\" and event_id=" << *event_id << " std::endl;
```

Reply:
Thanks! I've updated the code according to your suggestion.